tests/Unit/Domain/Creators/Test_Sphere.cpp

```diff
@@ -784,42 +771,121 @@ void test_kerr_horizon_conforming(const bool use_time_dependence) {
       std::move(time_dependent_options)};
   const auto domain = domain_creator.create_domain();
   const auto functions_of_time = domain_creator.functions_of_time();
-  // Set up coordinates on an ellipsoid of constant Boyer-Lindquist radius
-  const size_t num_points = 10;
-  MAKE_GENERATOR(gen);
-  std::uniform_real_distribution<double> dist_phi{0., 2. * M_PI};
-  std::uniform_real_distribution<double> dist_theta{0., M_PI};
-  const std::array<DataVector, 2> theta_phi{
-      {make_with_random_values<DataVector>(
-           make_not_null(&gen), make_not_null(&dist_theta), num_points),
-       make_with_random_values<DataVector>(
-           make_not_null(&gen), make_not_null(&dist_phi), num_points)}};
-  const auto radius =
-      get(gr::Solutions::kerr_schild_radius_from_boyer_lindquist(
-          inner_radius, theta_phi, mass, spin));
-  tnsr::I<DataVector, 3> x{};
-  get<0>(x) = radius * sin(get<0>(theta_phi)) * cos(get<1>(theta_phi));
-  get<1>(x) = radius * sin(get<0>(theta_phi)) * sin(get<1>(theta_phi));
-  get<2>(x) = radius * cos(get<0>(theta_phi));
   // Map the coordinates through the domain. They should lie at the lower zeta
   // boundary of their block.
   const auto x_logical =
-      block_logical_coordinates(domain, x, 0., functions_of_time);
-  for (size_t i = 0; i < num_points; ++i) {
+      block_logical_coordinates(domain, x, time, functions_of_time);
+  for (size_t i = 0; i < get<0>(x).size(); ++i) {
     CAPTURE(x_logical[i]);
     REQUIRE(x_logical[i].has_value());
     CHECK(get<2>(x_logical[i]->data) == approx(-1.));
   }
 }
 
+template <typename Generator>
+void test_shape_distortion(const gsl::not_null<Generator*> gen) {
+  domain::creators::Sphere::TimeDepOptionType time_dependent_options{};
+
+  // Set up theta phis
+  const size_t num_points = 10;
+  std::uniform_real_distribution<double> dist_phi{0., 2. * M_PI};
+  std::uniform_real_distribution<double> dist_theta{0., M_PI};
+  const std::array<DataVector, 2> theta_phi{
+      {make_with_random_values<DataVector>(gen, make_not_null(&dist_theta),
+                                           num_points),
+       make_with_random_values<DataVector>(gen, make_not_null(&dist_phi),
+                                           num_points)}};
+
+  const double time = 0.7;
+  const size_t l_max = 32;
+  const double mass = 0.8;
+  const std::array<double, 3> spin{{0.0, 0.0, 0.9}};
+  const double r_plus = mass * (1. + sqrt(1. - dot(spin, spin)));
+  const double inner_radius = r_plus;
+  std::string info_string{
+      "Check that inner radius is deformed to constant Boyer-Lindquist radius"};
```

Comment:
Don't pass this to functions. Just use a scope and set `INFO("...")` for each test.